central/complianceoperator/v2/compliancemanager/manager_impl.go

```diff
@@ -236,11 +236,53 @@ func (m *managerImpl) HandleScanRequestResponse(ctx context.Context, requestID s
 	return nil
 }
 
-func (m *managerImpl) ProcessRescanRequest(_ context.Context, _ interface{}) error {
-	// TODO(ROX-18091):
+func (m *managerImpl) ProcessRescanRequest(ctx context.Context, scanID string) error {
+	if !features.ComplianceEnhancements.Enabled() {
+		return errors.Errorf("Compliance is disabled. Cannot run compliance scan for configuration with ID %s", scanID)
+	}
+
 	// 1. Validate config exists in database
-	// 2. Push request to Sensor
-	panic("implement me")
+	scanConfig, found, err := m.scanSettingDS.GetScanConfiguration(ctx, scanID)
+	if err != nil {
+		return errors.Errorf("Encountered error attempting to find scan configuration with ID: %s", scanID)
+	} else if !found {
+		return errors.Errorf("Failed to find scan configuration by ID: %s", scanID)
+	}
+
+	// 2. Find applicable clusters for the given scan configuration (or just broadcast?)
+	clusters, err := m.scanSettingDS.GetScanConfigClusterStatus(ctx, scanID)
+	if err != nil {
+		return errors.Errorf("Failed to find clusters using scan configuration ID: %s", scanID)
+	}
+
+	for _, cluster := range clusters {
+		sensorRequestID := uuid.NewV4().String()
+		sensorMessage := &central.MsgToSensor{
+			Msg: &central.MsgToSensor_ComplianceRequest{
+				ComplianceRequest: &central.ComplianceRequest{
+					Request: &central.ComplianceRequest_ApplyScanConfig{
+						ApplyScanConfig: &central.ApplyComplianceScanConfigRequest{
+							Id: sensorRequestID,
+							ScanRequest: &central.ApplyComplianceScanConfigRequest_RerunScan{
+								RerunScan: &central.ApplyComplianceScanConfigRequest_RerunScheduledScan{
+									ScanSettings: &central.ApplyComplianceScanConfigRequest_BaseScanSettings{
+										ScanName: "ocp4-" + scanConfig.GetScanName(),
```

Comment:
Removed the hardcoded `ocp4` bit here - I did this just to make sure the plumbing between central and sensor was working properly.